source/board/microbitv2/nrf52820/power.c

```diff
@@ -19,19 +19,190 @@
  * limitations under the License.
  */
 
+#include "nrf.h"
+
+#include "gpio.h"
+#include "gpio_extra.h"
+#include "main_interface.h"
+#include "pwr_mon.h"
+#include "IO_Config.h"
+#include "DAP_config.h"
+#include "uart.h"
+#include "i2c.h"
+#include "rl_usb.h"
+
 #include "power.h"
 
+
+static void power_before(bool systemoff);
+static void power_after();
+static void power_systemoff();
+static void power_wfi();
+
+void POWER_CLOCK_IRQHandler(void);
+void GPIOTE_IRQHandler(void);
+
+
+extern volatile uint8_t wake_from_reset;
+extern volatile uint8_t wake_from_usb;
+extern volatile bool usb_pc_connected;
+extern main_usb_connect_t usb_state;
+extern power_source_t power_source;
+
+
+uint8_t  power_in_wfi;
+uint8_t  power_gpiote_enabled;
+uint32_t power_gpiote_intenset;
+
+
 void power_init()
 {
-    // TODO
+    power_in_wfi = 0;
+
+    gpio_cfg_input(GPIO_REG(RESET_BUTTON), GPIO_IDX(RESET_BUTTON), RESET_BUTTON_PULL);
```

Comment:
Is the pull up required? Is there no external one?

Reply:
I followed https://github.com/mathias-arm/DAPLink/blob/feature/nrf52820_hic_rebase_extra/source/hic_hal/nordic/nrf52820/gpio.c#L48
I was in two minds about whether to set it here at all.

Reply:
I mean we can update this line here (and wherever the reset button gpio might be reconfigure in nrf52820-specific files) to configure the pin without the pull up.
And leave `gpio_init()` as it is, because that is generic for any other nrf-based port.

> Should the pin be configured during sleep for IRQ on GPIO_PIN_CNF_SENSE_Low?

Is that to wake up from sleep when the signal goes from high to low?
In that case I think that's right, but please check if that is what the KL27 is doing as well in the latest version.
We'll also need to update the DAPLink bootloader to not enter in MAINTENANCE mode when woken up via interrupt, as done here: https://github.com/mathias-arm/DAPLink/commit/37d3a53ea96e46ef738c218a090e5902d84d711f

Reply:
OK. I have updated all reset pin settings to use NOPULL and checked the sleep interrupt.
I have made a note to check the bootloader wake up.